src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -63,29 +63,76 @@ namespace Microsoft.Quantum.Convert {
     /// Converts a given big integer to an equivalent integer, if possible.
     /// The function returns a pair of the resulting integer and a Boolean flag
     /// which is true, if and only if the conversion was possible.
-    /// # Remarks
-    /// See [C# BigInteger constructor](https://docs.microsoft.com/dotnet/api/system.numerics.biginteger.-ctor?view=netframework-4.7.2#System_Numerics_BigInteger__ctor_System_Int64_) for more details.
     function MaybeBigIntAsInt(a : BigInt) : (Int, Bool) {
-        body intrinsic;
+        if a > (1L <<< 63) - 1L or a < (-1L * (1L <<< 63)) {
+            return (0, false);
+        }
+        let arr = BigIntAsBoolArray(a);
+        let len = Length(arr);
+
+        // BigIntAsBoolArray always returns padded results with minimum length 8, so the below
+        // logic can assume the last entry is the sign bit for two's complement.
+        mutable val = 0;
+        for i in 0..(len - 2) {
+            set val += arr[i] ? 2 ^ i | 0;
+        }
+        if arr[len - 1] {
+            set val -= 2 ^ (len - 1);
+        }
+
+        return (val, true);
     }
 
 
     /// # Summary
     /// Converts a given big integer to an array of Booleans.
     /// The 0 element of the array is the least significant bit of the big integer.
-    /// # Remarks
-    /// See [C# BigInteger constructor](https://docs.microsoft.com/dotnet/api/system.numerics.biginteger.-ctor?view=netframework-4.7.2#System_Numerics_BigInteger__ctor_System_Int64_) for more details.
     function BigIntAsBoolArray(a : BigInt) : Bool[] {
-        body intrinsic;
+        // To use two's complement, little endian representation of the integer, we fisrt need to track if the input
+        // is a negative number. If so, flip it back to positive and start tracking a carry bit.
+        let isNegative = a < 0L;
+        mutable carry = isNegative;
+        mutable val = isNegative ? -a | a;
+
+        mutable arr = [];
+        while val != 0L {
+            let newBit = val % 2L == 1L;
+            if isNegative {
+                // For negative numbers we must invert the calculated bit, so treat "true" as "0"
+                // and "false" as "1". This means when the carry bit is set, we want to record the
+                // calculated new bit and set the carry to the opposite, otherwise record the opposite
+                // of the calculate bit.
```

Comment:
I'll admit, the explanation here of two's complement is paraphrased from the Wikepedia page: https://en.wikipedia.org/wiki/Two%27s_complement. I assumed familiarity with two's complement when writing this, and rather than explain more broadly how that works would expect someone to look it up and get a baseline understanding before reading the further explanations here.